content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,122 @@
+
+`InTreePlugin{vendor}Unregister` is a standalone feature gate that can be enabled and disabled out of CSI Migration scope. The name speaks for itself, when enabled, the component will not register the specific in-tree storage plugin to the supported list. If the cluster operator only enables this flag, end users will get an error from PVC saying it cannot find the plugin when the plugin is used. The cluster operator may want to enable this regardless of CSI Migration if they do not want to support the legacy in-tree APIs and only support CSI moving forward.
+
+At the same time, `CSIMigration{provider}Complete` was deprecated and removed in kubernetes v1.21 as well. The `InTreePlugin{vendor}Unregister` will replace the old feature flag but remain all the functionality that `CSIMigration{provider}Complete` provides.
+
+### Observability
+
+We have added the [CSI Migration Metrics](https://github.com/kubernetes/kubernetes/issues/98279) support for better observability of the feature. This will also help us discover any potential issues regarding the feature.
+
+- storage_operation_duration_seconds
+  - This metric is exposed from the Kube-Controller-Manager. A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_sidecar_operations_seconds
+  - This metric is exposed from the CSI sidecars(csi-provisioner, csi-attacher, csi-resizer). A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_kubelet_operations_seconds
+  - This metric is exposed from Kubelet. It exposes CSI operation, driver name, status and `migrated` field to indicate whether this storage operation is a CSI migration operation.
+
+### Bug fixes and feature improvement
+
+We have been able to identify bugs like dangling attachment, garbage collection, incorrect topology label ... and we have fixed all of them to make the feature more healthy.
+
+### Cloud Provider && Cluster Lifecycle Collaboration
+
+We, SIG Storage, have been working closely with SIG CloudProvider and SIG ClusterLifeCycle to provide a better experience for the feature launching.
+
+If you are a user of public cloud provider managed Kubernetes service like GKE, EKS, AKS, there is nothing required to be done from your end. The cloud providers will manage the feature gate and ensure the feature lands correctly.
+
+If you are a user of some other kubernetes distribution services, check the official documentations to get support for the feature. For the CSI Migration feature graduation to GA, we are coordinating with SIG ClusterLifeCycle towards requesting an early planning & they have been working hard to ensure a smooth experience.
```

Comment:
If you use a distribution of Kubernetes, check its official documentation for information about support for ths feature. For the CSI Migration feature graduation to GA, SIG Storage and SIG Cluster Lifecycle are collaborating towards making the migration mechanisms available in tooling (such as `kubeadm`) as soon as they're available in Kubernetes itself.
```